dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnit.java

```diff
@@ -282,34 +287,55 @@ public void removeAllUsers()
         users.clear();
     }
 
-    public List<OrganisationUnit> getSortedChildren()
+    public List<OrganisationUnit> getSortedChildren( String sortBy )
     {
         List<OrganisationUnit> sortedChildren = new ArrayList<>( children );
 
+        if ( SORT_BY_SHORT_NAME.equalsIgnoreCase( sortBy ) )
+        {
+            Collections.sort( sortedChildren, OrganisationUnitShortNameComparator.INSTANCE );
+            return sortedChildren;
+        }
+
         Collections.sort( sortedChildren );
 
         return sortedChildren;
     }
 
+    public List<OrganisationUnit> getSortedChildren()
+    {
+        return getSortedChildren( SORT_BY_NAME );
+    }
+
     public static List<OrganisationUnit> getSortedChildren( Collection<OrganisationUnit> units )
+    {
+        return getSortedChildren( units, SORT_BY_NAME );
+    }
+
+    public static List<OrganisationUnit> getSortedChildren( Collection<OrganisationUnit> units, String sortBy )

```

Comment:
Can we instead of a string sortBy use DisplayProperty.java ? That will make it "typed" and less error prone.